enhancements/siteconfig-extra-manifest-version-independence.md

```diff
@@ -0,0 +1,339 @@
+
+* **Currently** customer can put custom manifests in the git repo and point them via **extraManifestPath**. 
+* **Current behavior:** If customer only want to include custom CRs to build site specific CRs:
+  * **inclusionDefault** must be set to *exclude* which will exclude all the CRs from ztp-container
+  * pass the list of CR file names under **filter.include**, that will include listed CRs.
+  * but it does not allow to have the [same CR name](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go#L462-L464) for the custom CRs. 
+
+  * to be able to add custom CRs, customer needs to change the name of the custom CR.
+
+
+  ```yaml
+  - cluster:
+    extraManifestPath: mypath/
+    extraManifests:
+      filter:
+        inclusionDefault: exclude
+        include:
+          - myCR.yaml					
+  ```
+
+  * as a result same named CR coming from user directory doesn't take precedent, rather siteconfig issues an error complaining same named CR is not allowed.
+
+### Goals
+
+- Allow customer to include any custom CRs having same/different name than builtIn CRs residing in the container
+- Provide a better control on how to enable and disable builtIn CRs
+- Keep backward compatibility of the behavior corresponds to 4.13 version.
+
+### Non-Goals
+
+- TBD
+### User Stories
+
+- [EPIC LINK](https://issues.redhat.com/browse/CNF-7365) --> [User Story](https://issues.redhat.com/browse/CNF-8672)
+
+## Proposal
+
+* **Controllability:** The proposal is to introduce a parameter called  <span style="color:red">*extraManifests.builtinCRs*</span>. which can hold value "*include|exclude*", which provide the flexibility for the customer to include/exclude CRs that is provided via ztp-container. 
+
+  * In this way, customer have full control of what CRs goes into the siteconfig. Until 4.13 release, customer couldn't put same named CR to the CR provided via ztp-container in the *extraManifestPath*. 
+
+
+ ```yaml
+  - cluster:
+    extraManifestPath: mypath/
+    extraManifests:
+      builtinCRs: include | exclude   <----- New field
+      filter:
+        inclusionDefault: exclude
+        include:
+          - myCR.yaml					
+  ```
+
+* **Version Independence**: This change will allow customer to have version independence of extra-manifests having different versioned folder in their git repository. Customer can extract versioned extra-manifests and can push to their git repo. An example has shown below:
+
+
+```yaml
+siteconfig/
+├── kustomization.yaml
+├── version_4.12
+│   ├── kustomization.yaml
+│   ├── sno-extra-manifest/
+│   ├── cnfde9.yaml
+└── version_4.13
+    ├── kustomization.yaml
+    ├── sno-extra-manifest/
+    ├── cnfde10.yaml    
+```
+
+### Workflow Description
+
+- Due to the proposal, now we have different combination of *include|exclude* in extraManifests and the filter level. Overall 4 scenarios are briefly explained in the below table based on the proposal: 
```

Comment:
Could you expand the examples a little with input CRs and output CRs? 
e.g in case 1
```
    extraManifestPath: mypath/
    extraManifests:
      builtinCRs: exclude
      filter:
        inclusionDefault: exclude
        include:
          - myCR.yaml					
```
We can have 2 sets:
- ztp-extra-manifests [ztpCR1.yaml, ztpCR2.yaml]
- custom-extra-manifests [myCR.yaml, myCR2.yaml]

So final output in the CM would contain: [myCR.yaml] ?

Reply:
Example of final output added with commit 5fbd818231476ca3ff8fcf03632ad6e9ba81110b